frame/support/src/executor.rs

```diff
@@ -0,0 +1,620 @@
+// This file is part of Substrate.
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use sp_std::prelude::*;
+use crate::{weights::Weight, traits::Get};
+use codec::{Encode, Decode};
+use sp_runtime::traits::Zero;
+use crate::{RuntimeDebugNoBound, PartialEqNoBound, EqNoBound, CloneNoBound};
+
+const LOG_TARGET: &'static str = "runtime::task_executor";
+
+/// A task that can be stored in storage and executed at some later time.
+///
+/// This trait itself does not make any assumptions about *when* the task is executed. As far as
+/// this trait is concerned, it can be now, all at once. It can be executed as mandatory work in
+/// `on_initialize` or `on_finalize`, or in some other low priority circumstance (e.g. on_idle).
+///
+/// If the type implementing this trait is generic over `<T: Config>` then one needs to derive
+/// [`CloneNoBound`], [`PartialEqNoBound`], [`EqNoBound`], [`RuntimeDebugNoBound`], as opposed to
+/// their normal counterparts, and implement `Default` manually. This is to prevent `T` to be
+/// bounded to these traits.
+pub trait RuntimeTask:
+	Sized + Clone + Default + Encode + Decode + PartialEq + Eq + sp_std::fmt::Debug
```

Comment:
Why do we need `Default`?

Reply:
I believe everything going to storage needs `Default`.

Reply:
But the `RuntimeTask` does not go into storage. The `StoredExecutor` does.